core/chains/evm/logpoller/log_poller.go

```diff
@@ -57,8 +57,31 @@ type LogPoller interface {
 	IndexedLogsTopicRange(eventSig common.Hash, address common.Address, topicIndex int, topicValueMin common.Hash, topicValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error)
 	IndexedLogsWithSigsExcluding(address common.Address, eventSigA, eventSigB common.Hash, topicIndex int, fromBlock, toBlock int64, confs int, qopts ...pg.QOpt) ([]Log, error)
 	LogsDataWordRange(eventSig common.Hash, address common.Address, wordIndex int, wordValueMin, wordValueMax common.Hash, confs int, qopts ...pg.QOpt) ([]Log, error)
```

Comment:
Any reason to not add finalized log query support to all the queries? I think there are other places in CCIP we want it (and other products also may want that)

Reply:
If we decide to go with `type Confirmations int`, then we should definitely support it in every query. If we decide to go with the struct then I was worried about impacting other products using LogPoller, as it breaks the backward compatibility